### PR TITLE
Support for doc & url for required module

### PR DESCRIPTION
**Pull request by angelozerr**
This PR looks like findTypeAt of node tern plugin, to support doc and url when find type is done for required module.

### CODE REVIEW

File: plugin/requirejs.js [hunk 1]
```diff
@@ -255,19 +255,26 @@
...
-    // Provide a custom origin location pointing to the require()d file
-    var exportedType = expr.node.required;
-    type.origin = exportedType.origin;
-    type.originNode = exportedType.originNode;
+    if (!expr) return type;
+    var isStringLiteral = expr.node.type === "Literal" &&
+       typeof expr.node.value === "string";
+    var isRequireArg = !!expr.node.required;
+
+    if (isStringLiteral && isRequireArg) {
```
Comment (marijnh) [inline plugin/requirejs.js]: Why the apparently random refactor to use boolean variables and an extra condition rather than the existing early return?

Comment (angelozerr) [inline plugin/requirejs.js]: I have copied/pasted the code from node tern plugin and adapt it for requirejs. If you don't like this code, I think we should modify node tern plugin too.

**Comment by mdasrafulalamasa** [comment]
My resume and the first

File: plugin/requirejs.js (hunk 1 above)
Comment (angelozerr) [inline plugin/requirejs.js]: @marijnh do you want a new PR for that?

**Comment by marijnh** [comment]
Does attached two-line patch do what you needed?

**Comment by angelozerr** [comment]
Yes thanks @marijnh. I close this PR.

**Pull request closed without merging**